app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import DashboardHelpers from '../DashboardHelpers';
+import StudentsTable from '../StudentsTable';
+import DashboardBarChart from '../DashboardBarChart';
+import DateSlider from '../DateSlider';
+
+class SchoolDisciplineDashboard extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: DashboardHelpers.schoolYearStart(),
+      selectedChart: 'location',
+      selectedCategory: null
+    };
+    this.setDate = this.setDate.bind(this);
+    this.setStudentList = this.setStudentList.bind(this);
+    this.resetStudentList = this.resetStudentList.bind(this);
+    this.selectChart = this.selectChart.bind(this);
+  }
+
+  setDate(range) {
+    this.setState({
+      startDate: moment.unix(range[0]).format("YYYY-MM-DD")
+    });
+  }
+  setStudentList(highchartsEvent) {
+    this.setState({selectedCategory: highchartsEvent.point.category});
+  }
+  resetStudentList() {
+    this.setState({selectedCategory: null});
+  }
+  selectChart(event) {
+    this.setState({selectedChart: event.target.value, selectedCategory: null});
+  }
+
+  filterIncidentDates(incidents) {
+    return incidents.filter((incident) => {
+      return moment.utc(incident.occurred_at).isSameOrAfter(moment.utc(this.state.startDate));
+    });
+  }
+
+  studentDisciplineIncidentCounts(incidentCategory) {
+    let studentDisciplineIncidentCounts = {};
+    const selectedChart = this.getChartData(this.state.selectedChart);
```

Comment:
On the left side of this assignment `selectedChart` is an object, but on the right side `selectedChart` is a string.

Reply:
@alexsoble same as below, this should have been changed with an earlier refactor. Thanks!